File: src/main/java/vorquel/mod/vderps/item/ItemMultiVD.java
package vorquel.mod.vderps.item;

import java.util.List;

import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.util.IIcon;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

public class ItemMultiVD extends ItemVD {
	
	private String[] subname;
	@SideOnly(Side.CLIENT)
	protected IIcon[] itemIcons;
	
	public ItemMultiVD(String name, String... params) {
		super(name);
		this.subname = params;
	}
	
	@Override
	public String getUnlocalizedName(ItemStack itemStack) {
		int index = itemStack.getItemDamage();
		String sub = null;
		if(index < subname.length)
			sub = subname[index];
		return String.format("%s.%s", getUnlocalizedName(), sub);
	}
	
	@Override
	@SideOnly(Side.CLIENT)
	public void registerIcons(IIconRegister iconRegister) {
		itemIcons = new IIcon[subname.length];
		for(int i=0; i<subname.length; ++i) {
			itemIcons[i] = iconRegister.registerIcon(String.format("%s.%s", getShortName(), subname[i]));
		}
	}
	
	@Override
	@SideOnly(Side.CLIENT)
	public void getSubItems(Item item, CreativeTabs tab, List list) {
		for(ItemStack stack : getSubItemStacks())
			list.add(stack);
	}
	
	@Override
	@SideOnly(Side.CLIENT)
	public IIcon getIconFromDamage(int damage) {
		return itemIcons[damage];
	}
	
	public ItemStack[] getSubItemStacks(int stackSize) {
		ItemStack[] stacks = new ItemStack[subname.length];
		for(int i=0; i<subname.length; ++i)
			stacks[i] = new ItemStack(this, stackSize, i);
		return stacks;
	}
	
	public ItemStack[] getSubItemStacks() {
		return getSubItemStacks(1);
	}
}

File: src/main/scala/vorquel/mod/vderps/item/ItemMultiVD.scala
package vorquel.mod.vderps.item

import java.util

import cpw.mods.fml.relauncher.{SideOnly, Side}
import net.minecraft.client.renderer.texture.IIconRegister
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.{Item, ItemStack}
import net.minecraft.util.IIcon

class ItemMultiVD(name: String, params: String*) extends ItemVD(name) {
  private val subname: Seq[String] = params
  @SideOnly(Side.CLIENT)
  protected val itemIcons: Array[IIcon] = new Array(subname.length)

  override def getUnlocalizedName(itemStack: ItemStack): String = {
    val index: Int = itemStack.getItemDamage
    var sub: String = null
    if(index < subname.length) sub = subname(index)
    s"${getUnlocalizedName()}.$sub"
  }

  @SideOnly(Side.CLIENT) override def registerIcons(iconRegister: IIconRegister) {
    for(i <- 0 until subname.length)
      itemIcons(i) = iconRegister.registerIcon(s"$getShortName.${subname(i)}")
  }

  @SideOnly(Side.CLIENT)
  override def getSubItems(item: Item, tab: CreativeTabs, list: util.List[_]) {
    val temp = list.asInstanceOf[util.ArrayList[Object]]
    for(i <- 0 until subname.length)
      temp.add(new ItemStack(this, 1, i))
  }

  @SideOnly(Side.CLIENT)
  override def getIconFromDamage(damage: Int): IIcon = itemIcons(damage)
}
